packages/keychain/src/mnemonic/index.ts

```diff
@@ -0,0 +1,35 @@
+import { generateMnemonic as generateBip39Mnemonic, mnemonicToSeed } from 'bip39';
+import { ChainID } from '@blockstack/stacks-transactions';
+import { randomBytes } from 'blockstack/lib/encryption/cryptoRandom';
+import { bip32 } from 'bitcoinjs-lib';
+import { encrypt } from '../encryption/encrypt';
+import { encryptMnemonic } from 'blockstack/lib/encryption/wallet';
+
+export type AllowedKeyEntropyBits = 128 | 256;
+
+export async function generateMnemonicRootKeychain(
+  password: string,
+  entropy: AllowedKeyEntropyBits
+) {
+  const backupPhrase = generateBip39Mnemonic(entropy, randomBytes);
+  const seedBuffer = await mnemonicToSeed(backupPhrase);
+  const rootNode = bip32.fromSeed(seedBuffer);
+  const ciphertextBuffer = await encrypt(backupPhrase, password);
+  const encryptedMnemonicPhrase = ciphertextBuffer.toString('hex');
+  return {
+    rootNode,
+    encryptedMnemonicPhrase,
+  };
+}
+
+export async function deriveRootKeychainFromMnemonic(mnemonic: string, password: string) {
+  const encryptedMnemonic = await encrypt(mnemonic, password);
+  const encryptedMnemonicHex = encryptedMnemonic.toString('hex');
+  const seedBuffer = await mnemonicToSeed(mnemonic);
+  const rootNode = bip32.fromSeed(seedBuffer);
+  return {
+    rootNode,
+    encryptMnemonic,
```

Comment:
I'm not sure why this is necessary? The encrypt function actually just uses `encryptMnemonic` under the hood. It used be built in this way (i.e. a separate file for just the `encrypt` function) to make it usable in a WebWorker, but we don't need that anymore, since we've updated the underlying crypto libraries to be asynchronous.

Reply:
This was supposed to be `encryptedMnemonic`.